src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java

```diff
@@ -0,0 +1,369 @@
+    Keycloak keycloak = getAdminClient();
+    ClientRepresentation client = getClient(keycloak);
+    String realmName = getRealmName();
+    
+    if (client == null) {
+      throw new AuthorizationException(AUTHORIZATION_EXCEPTION_MESSAGE);
+    }
+    
+    RealmResource realm = keycloak.realm(realmName);
+    ScopePermissionsResource scopeResource = realm.clients().get(client.getId()).authorization().permissions().scope();
+    ScopePermissionRepresentation existingPermission = scopeResource.findByName(name);
+
+    ScopePermissionRepresentation representation = new ScopePermissionRepresentation();
+    representation.setResources(Collections.singleton(resourceId.toString()));
+    representation.setDecisionStrategy(decisionStrategy);
+    representation.setName(name);
+    representation.setScopes(scopes.stream().map(AuthorizationScope::getName).collect(Collectors.toSet()));
+    representation.setPolicies(Collections.singleton(policyId.toString()));
+
+    String result = representation.getName();
+    Response response = scopeResource.create(representation);
+    try {
+      
+      if (existingPermission == null) {
+        int status = response.getStatus();
+        
+        if (status != 201) {
+          String message = "Unknown error";
+          
+          try {
+            message = IOUtils.toString((InputStream) response.getEntity(), StandardCharsets.UTF_8);
+          } catch (IOException e) {
+            logger.warn("Failed read error message", e);
+          }
+
+          logger.warn("Failed to create scope permission for resource {} with message {}", resourceId, message);
+        }
+        
+      } else {
+        realm.clients().get(client.getId()).authorization().permissions().scope().findById(existingPermission.getId()).update(representation);
+      }
+      
+    } finally {
+      response.close();
+    }
+    
+    return result;
+  }
+
+  /**
+   * Assures that logged user has an existing policy
+   * 
+   * @param keycloak admin client
+   * @param realmName realm name
+   * @param clientId client id
+   * @return UUID user policy id
+   */
+  public UUID ensureUserPolicyExists(UUID userId) throws AuthorizationException {
+    String realmName = getRealmName();
+    Keycloak keycloak = getAdminClient();
+    RealmResource realm = keycloak.realm(realmName);
+    UsersResource users = realm.users();
+    ClientRepresentation client = getClient(keycloak);
+    String userIdString = userId.toString();
+    
+    if (client == null) {
+      throw new AuthorizationException(AUTHORIZATION_EXCEPTION_MESSAGE);
+    }
+    
+    UserPoliciesResource userPolicies = realm.clients().get(client.getId()).authorization().policies().user();
+    UserResource loggedUserResource = users.get(userIdString);
+    
+      String userName = loggedUserResource.toRepresentation().getUsername();
+      String policyNameTemplate = String.format("user-%s", userName);
+      UUID result = userId;
```

Comment:
Now the result is actually userId not the policy id the documentation claims it to be